v1/src/main/java/com/google/cloud/teleport/templates/VectorEmbeddingExport.java

```diff
@@ -0,0 +1,213 @@
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.View;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Dataflow template which export vector embeddings from Spanner to GCS in json format. It exports a
+ * Spanner table using <a
+ * href="https://cloud.google.com/spanner/docs/reads#read_data_in_parallel">Batch API</a>, which
+ * creates multiple workers in parallel for better performance. The result is written to a JSON file
+ * in Google Cloud Storage.
+ *
+ * <p>Check out <a
+ * href="https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v1/README_Spanner_to_GCS_Text.md">README</a>
+ * for instructions on how to use or modify this template.
+ */
+@Template(
+    name = "Cloud_Spanner_to_Vector_Embedding",
+    category = TemplateCategory.BATCH,
+    displayName = "Cloud Spanner to Vector Embeddings Cloud Storage",
+    description =
+        "A pipeline which reads vector embeddings from Cloud Spanner table and writes it to Cloud Storage as JSON"
+            + " files.",
+    optionsClass = SpannerToJsonOptions.class,
+    documentation =
+        "https://cloud.google.com/dataflow/docs/guides/templates/provided/cloud-spanner-to-cloud-storage",
+    contactInformation = "https://cloud.google.com/support")
+public class VectorEmbeddingExport {
+
+  private static final Logger LOG = LoggerFactory.getLogger(VectorEmbeddingExport.class);
+
+  /** Custom PipelineOptions. */
+  public interface SpannerToJsonOptions
+      extends PipelineOptions, SpannerReadOptions, FilesystemWriteOptions {
+
+    @TemplateParameter.GcsWriteFolder(
+        order = 1,
+        optional = true,
+        description = "Cloud Storage temp directory for storing JSON files",
+        helpText = "The Cloud Storage path where the temporary JSON files can be stored.",
+        example = "gs://your-bucket/your-path")
+    ValueProvider<String> getJSONTempDirectory();
+
+    @SuppressWarnings("unused")
+    void setJSONTempDirectory(ValueProvider<String> value);
+
+    @TemplateParameter.Enum(
+        order = 2,
+        enumOptions = {
+          @TemplateEnumOption("LOW"),
+          @TemplateEnumOption("MEDIUM"),
+          @TemplateEnumOption("HIGH")
+        },
+        optional = true,
+        description = "Priority for Spanner RPC invocations",
```

Comment:
Do we need to set default here or is it set in the ReadOptions?

Reply:
It's set to Medium for all the dataflow templates.